logstream errors abort all tests

/area API
/area test-and-release
/kind bug
/kind cleanup

If you look at [this](https://prow.knative.dev/view/gcs/knative-prow/logs/ci-knative-serving-contour-latest/1293407862143324160) continuous build, you will see that every single failure (that I clicked through) failed with the following:

```
kubelogs.go:236: Error during logstream error container "autoscaler-hpa" in pod "autoscaler-hpa-8c5988c8d-shw6x" is terminated
```

This is because once `k.err` is set [here](https://github.com/knative/pkg/blob/530eac8757df9c6fb33fd4a8b0c72a72fa909cc4/test/logstream/kubelogs.go#L84) every single "cancel" function we return will error [here](https://github.com/knative/pkg/blob/530eac8757df9c6fb33fd4a8b0c72a72fa909cc4/test/logstream/kubelogs.go#L236).  I believe this is similarly true for tests that hadn't even started when the pod was killed!

Given that this is purely diagnostic, I'm going to suggest that we bias heavily towards `t.Log` for this sort of error instead of using `t.Error` because it's entirely plausible that there's nothing really wrong.

cc @markusthoemmes @vagababov @dprotaso for thoughts

--- a/test/logstream/kubelogs.go
+++ b/test/logstream/kubelogs.go
@@ -41,7 +41,6 @@ type kubelogs struct {
 	once sync.Once
 	m    sync.RWMutex
 	keys map[string]logger
-	err  error
 }
 
 type logger func(string, ...interface{})
@@ -78,11 +77,7 @@ func (k *kubelogs) startForPod(pod *corev1.Pod) {
 			req := k.kc.Kube.CoreV1().Pods(psn).GetLogs(pn, options)
 			stream, err := req.Stream()
 			if err != nil {
-				func() {
-					k.m.Lock()
-					defer k.m.Unlock()
-					k.err = err
-				}()
+				k.handleGenericLine([]byte(err.Error()), pn)
 				return
 			}
 			defer stream.Close()
@@ -231,9 +226,5 @@ func (k *kubelogs) Start(t test.TLegacy) Canceler {
 		k.m.Lock()
 		defer k.m.Unlock()
 		delete(k.keys, name)
-
-		if k.err != nil {
-			t.Error("Error during logstream", "error", k.err)
-		}
 	}
 }

